static/src/views/Chat.vue

```diff
@@ -203,8 +221,18 @@ export default {
     async fetchMessages(){
       try {
         this.messagesLoaded = false;
+        // Not ideal, but basically we're going to abuse the plugin again.
+        // When we enter the flagged messages room, we are going to tell the plugin
+        // that there are no messages so that it tries to display the "messages empty"
+        // text, but we hijack it in the template to display a custom component
+        // that uses the flaggedMessages list to create a table of flagged messages.
+        const isFlaggedMessages = (this.roomId === this.flaggedMsgsId);
+        this.showFooter = !isFlaggedMessages;
 
-        const response = await MessageService.fetchMessages(this.roomId, this.offset, this.batchSize);
+        const response = isFlaggedMessages ? 
+                await MessageService.fetchFlaggedMessages({}) : 
```

Comment:
not related to this change, but I just realized that the message service will only ever return one batch of messages (so if there are more than 50 flagged it'll be impossible to see them). Is that an acceptable state for shipping?

Reply:
Good catch!

I actually looked into that last night. The MessageService currently doesn't accept a limit or offset (https://github.com/dimagi/mpact/blob/5d12372ef5cee26d07756d27e5acce3f0c9e2266/static/src/services/MessageService.js#L83), which in the Django code means it returns all of the records at once (https://github.com/dimagi/mpact/blob/5d12372ef5cee26d07756d27e5acce3f0c9e2266/mpact/services.py#L202)

It's not intuitive that the two APIs behave differently (or even that there are two APIs for fetching messages), but I just left it at that.

Reply:
oh nice, ha. I just looked at the function declaration and assumed it would just set defaults if they weren't specified.

I suppose that's probably fine as long as we don't expect thousands of flagged messages anytime soon.